### PR TITLE
Application v1.0

### PR DESCRIPTION
**Pull request by Valelaaa**

### CODE REVIEW

File: src/main/java/com/endava/endavapp/entity/Employee.java [hunk 1]
```diff
@@ -45,10 +45,9 @@ public class Employee {
     private String lastName;
 
 
-    @ManyToOne
-    @EqualsAndHashCode.Exclude
+    @ManyToOne(targetEntity = Department.class)
     @JoinColumn(name = "department_id")
-    private Department department;
+    private Department departmentId;
```
Comment (NataliaIovva) [inline src/main/java/com/endava/endavapp/entity/Employee.java:50]: rename variable to department

File: src/main/java/com/endava/endavapp/repository/EmployeeRepository.java [hunk 2]
```diff
@@ -5,10 +5,9 @@
 import org.springframework.stereotype.Repository;
 
 import java.util.Optional;
-import java.util.UUID;
 
 @Repository
-public interface EmployeeRepository extends JpaRepository<Employee, UUID> {
+public interface EmployeeRepository extends JpaRepository<Employee, String> {
     Optional<Employee> findByEmail(final String email);
     Optional<Employee> findByPhoneNumber(final String email);
```
Comment (NataliaIovva) [inline src/main/java/com/endava/endavapp/repository/EmployeeRepository.java:12]: `Optional<Employee> findByPhoneNumber(final String phoneNumber);`

File: src/main/java/com/endava/endavapp/service/EmployeeServiceImpl.java [hunk 3]
```diff
@@ -81,7 +88,7 @@ public EmployeeDto editEmployee(final EmployeeDto employeeDTO, final UUID employ
             updateEmployee.setLastName(employeeDTO.getLastName());
             updateEmployee.setPhoneNumber(employeeDTO.getPhoneNumber());
             updateEmployee.setSalary(employeeDTO.getSalary());
-            updateEmployee.setDepartment(departmentRepository
+            updateEmployee.setDepartmentId(departmentRepository
```
Comment (NataliaIovva) [inline src/main/java/com/endava/endavapp/service/EmployeeServiceImpl.java:91]:
```
final String departmentName = employeeDTO.getDepartment().getDepartmentName(); 

final Department department =departmentRepository.findFirstByDepartmentName(departmentName)
                                                    .orElse(null); 
updateEmployee.setDepartmentId(department );

File: src/main/java/com/endava/endavapp/service/EmployeeServiceImpl.java [hunk 4]
```diff
@@ -50,25 +50,32 @@ public boolean addEmployee(final EmployeeDto employeeDTO) {
...
         );
+    }
+    @Override
+    public boolean addEmployee(final EmployeeDto employeeDTO) {
+        addValidation(employeeDTO);
         final Department department = departmentRepository
-                .findByDepartmentNameAndAndLocation(employeeDTO.
-                                getDepartment().getDepartmentName(),
-                        employeeDTO.getDepartment().getLocation()).orElse(null);
+                .findByDepartmentNameAndLocation(employeeDTO
```
Comment (NataliaIovva) [inline src/main/java/com/endava/endavapp/service/EmployeeServiceImpl.java:58]: please reformat code

File: src/main/java/com/endava/endavapp/service/DepartmentServiceImpl.java [hunk 5]
```diff
@@ -30,33 +32,70 @@ public List<DepartmentDto> getAll() {
...
+                departmentDto.getDepartmentName(),
+                departmentDto.getLocation());
+
+        final String departmentId =  departmentRepository
+                .findFirstByDepartmentName(name)
+                .orElse(null)
+                .getDepartmentId();
+
+        final Department updateDepartment =
+                departmentRepository.getReferenceById(departmentId);
```
Comment (NataliaIovva) [inline src/main/java/com/endava/endavapp/service/DepartmentServiceImpl.java:93]: please reformat code

File: src/main/java/com/endava/endavapp/entity/Employee.java (hunk 1 above)
Comment (Valelaaa) [reply src/main/java/com/endava/endavapp/entity/Employee.java:50]: Done

File: src/main/java/com/endava/endavapp/repository/EmployeeRepository.java (hunk 2 above)
Comment (Valelaaa) [reply src/main/java/com/endava/endavapp/repository/EmployeeRepository.java:12]: Done

File: src/main/java/com/endava/endavapp/service/EmployeeServiceImpl.java (hunk 3 above)
Comment (Valelaaa) [reply src/main/java/com/endava/endavapp/service/EmployeeServiceImpl.java:91]: Done

File: src/main/java/com/endava/endavapp/service/EmployeeServiceImpl.java (hunk 4 above)
Comment (Valelaaa) [reply src/main/java/com/endava/endavapp/service/EmployeeServiceImpl.java:58]: Done

File: src/main/java/com/endava/endavapp/service/DepartmentServiceImpl.java (hunk 5 above)
Comment (Valelaaa) [reply src/main/java/com/endava/endavapp/service/DepartmentServiceImpl.java:93]: Done

File: src/main/java/com/endava/endavapp/exeption/AlreadyExistException.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package com.endava.endavapp.exeption;
```
Comment (NataliaIovva) [inline src/main/java/com/endava/endavapp/exeption/AlreadyExistException.java:1]: `package com.endava.endavapp.exception;`

File: src/main/java/com/endava/endavapp/exeption/ElementNotFoundException.java [hunk 7]
```diff
@@ -0,0 +1,7 @@
+package com.endava.endavapp.exeption;
```
Comment (NataliaIovva) [inline src/main/java/com/endava/endavapp/exeption/ElementNotFoundException.java:1]: `package com.endava.endavapp.exception;`

**Review by NataliaIovva — APPROVED** [review]

**Pull request merged**